IMS.BL/Application/Program.cs

```diff
@@ -0,0 +1,104 @@
+﻿using System;
+using System.Linq;
+using IMS.BL.Database.DatabaseConnections.SqlDatabaseConnection;
+using IMS.BL.Domain;
+using IMS.BL.Repositories;
+
+namespace IMS.BL.Application
+{
+    internal class Program
+    {
+        public static void Main(string[] args)
+        {
+            Console.WriteLine("Please choose what operation you want to perform.");
+            Console.WriteLine("1 --> Add New Product.");
+            Console.WriteLine("2 --> Edit Existing Product.");
+            Console.WriteLine("3 --> Delete Existing Product.");
+            Console.WriteLine("4 --> Search For A Product.");
+            Console.WriteLine("5 --> Get All Products.");
+            Console.WriteLine("0 --> Exit.");
+            
+            int choice;
+            while (!int.TryParse(Console.ReadLine(), out choice))
+            {
+                Console.WriteLine("Please, enter a valid number from the choices above.");
+            }           
+            
+            // Required objects.
+            var sqlConnection = SqlConnectionProvider.Instance.SqlConnectionObject;
+            var sqlInventoryRepository = new SqlInventoryRepository(sqlConnection);
+            var inventoryRepository = new InventoryRepository();
+            inventoryRepository.SetInventoryRepository(sqlInventoryRepository);
```

Comment:
Why setter for the repository? make a constructor and pass it through.

Reply:
I thought this would be the best approach to do it, but it'll require another instance if I want to change the main strategy/repository. So either a constructor and a setter method or just a setter method can work.

Reply:
This is not a good approach. Lets say you didnt set the inventoryRepositroy, then when you try to AddNewProduct for example it will throw NullPointer since you didnt set it. Constructors are there to show us what is the contract of the object and what it needs to be instantiated. This way we can't instantiate InventoryService unless InventoryRpeository is passed in the constructor